MonoGame.Framework/Graphics/GraphicsDevice.cs

```diff
@@ -262,6 +278,35 @@ public DepthStencilState DepthStencilState
             }
         }
 
+        internal void ApplyState(bool applyShaders)
+        {
+            PlatformBeginApplyState();
+
+            if (_blendStateDirty)
+            {
+                // Static state properties never actually get bound;
+                // instead we use our GraphicsDevice-specific version of them.
+                var newBlendState = _blendState;
+                if (ReferenceEquals(_blendState, BlendState.Additive))
+                    newBlendState = BlendStateAdditive;
+                else if (ReferenceEquals(_blendState, BlendState.AlphaBlend))
+                    newBlendState = BlendStateAlphaBlend;
+                else if (ReferenceEquals(_blendState, BlendState.NonPremultiplied))
+                    newBlendState = BlendStateNonPremultiplied;
+                else if (ReferenceEquals(_blendState, BlendState.Opaque))
+                    newBlendState = BlendStateOpaque;

```

Comment:
Maybe all this code should be moved to the `BlendState` setter.  This way it only occurs once on set and not on every draw call.  This would mean you need to keep two temporaries....`_blendState` and `_realBlendState` or something.